src/engine/BMDie.php

```diff
@@ -423,13 +422,28 @@ public function roll($isTriggeredByAttack = FALSE, $isSubdie = FALSE) {
         if ($turboDieIsAboutToBeReplaced) {
             $this->value = NULL;
         } elseif ($needsToReroll) {
-            $this->set__value(bm_rand($this->min, $this->max));
+            $this->select_new_value();
         }
 
         $this->run_hooks('post_roll', array('die' => $this,
                                             'isTriggeredByAttack' => $isTriggeredByAttack));
     }
 
+    /**
+     * Select a new value
+     */
+    protected function select_new_value() {
+        $this->set__value(bm_rand($this->min, $this->max));
+    }
+
+    /**
+     * This function lets us specify the value that we want to end up with after rerolling,
+     * even when we're doing odd things like drawing cards for Wildcard.
+     */
+    public function force_specific_value($value) {
+        $this->value = $value;
+    }
```

Comment:
I don't love the word "force" used in a non-exceptional condition.  I think it's misleading to future devs.  Why *aren't* we just calling this `set_value()` (or even `set_specific_value()` if you like, but i think the fact that the value we're setting is being passed as an argument implies "specific").

Reply:
I think this is an exceptional condition. The inherited method for Wildcard is used for fire turndown, and it means that we continue drawing from the deck until we find a card with the correct value. That's quite different from just setting the value directly.